ql/src/java/org/apache/hadoop/hive/ql/metadata/client/SessionMetaStoreClientProxy.java

```diff
@@ -0,0 +1,3242 @@
+    } else if (sourceTempTable != null && destTempTable != null) {
+      return exchangePartitions(partitionSpecs, sourceTempTable, getPartitionedTempTable(sourceTempTable),
+          destTempTable, getPartitionedTempTable(destTempTable));
+    }
+    throw new MetaException("Exchanging partitions between temporary and non-temporary tables is not supported.");
+  }
+
+  @Override
+  public void alter_partition(String catName, String dbName, String tblName, Partition newPart,
+      EnvironmentContext environmentContext) throws TException {
+    alter_partition(catName, dbName, tblName, newPart, environmentContext, null);
+  }
+
+  @Override
+  public void alter_partition(String catName, String dbName, String tblName, Partition newPart,
+      EnvironmentContext environmentContext, String writeIdList)
+      throws TException {
+    Table table = getTempTable(dbName, tblName);
+    if (table == null) {
+      getDelegate().alter_partition(catName, dbName, tblName, newPart, environmentContext, writeIdList);
+      return;
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    tt.alterPartition(newPart);
+  }
+
+  @Override
+  public void alter_partitions(String catName, String dbName, String tblName, List<Partition> newParts,
+      EnvironmentContext environmentContext, String writeIdList, long writeId) throws TException {
+    Table table = getTempTable(dbName, tblName);
+    if (table == null) {
+      getDelegate().alter_partitions(catName, dbName, tblName, newParts, environmentContext, writeIdList, writeId);
+      return;
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    tt.alterPartitions(newParts);
+  }
+
+  @Override
+  public void renamePartition(String catName, String dbname, String tableName, List<String> partitionVals,
+      Partition newPart, String validWriteIds, long txnId, boolean makeCopy) throws TException {
+    Table table = getTempTable(dbname, tableName);
+    if (table == null) {
+      getDelegate().renamePartition(catName, dbname, tableName, partitionVals, newPart, validWriteIds, txnId, makeCopy);
+      return;
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    tt.renamePartition(partitionVals, newPart);
+  }
+
+  @Override
+  public Partition appendPartition(String catName, String dbName, String tableName, List<String> partVals)
+      throws TException {
+    Table table = getTempTable(dbName, tableName);
+    if (table == null) {
+      return getDelegate().appendPartition(catName, dbName, tableName, partVals);
+    }
+    if (partVals == null || partVals.isEmpty()) {
+      throw new MetaException("The partition values must be not null or empty.");
+    }
+    assertTempTablePartitioned(table);
+    Partition partition = new PartitionBuilder().inTable(table).setValues(partVals).build(conf);
+    return appendPartitionToTempTable(table, partition);
+  }
+
+  @Override
+  public Partition appendPartition(String catName, String dbName, String tableName, String partitionName)
```

Comment:
We have to override `appendPartition(String dbName, String tableName, String name)`